pycozmo/anim.py

```diff
@@ -135,3 +135,40 @@ def play(self, cli):
                 time.sleep(delay_ms)
 
         cli.conn.send(protocol_encoder.EndAnimation())
+
+
+class AnimationGroupMember:
+
+    __slots__ = [
+        "name",
+        "weight",
+        "cooldown_time",
+        "mood",
+    ]
+
+    def __init__(self, name: str, weight: float, cooldown_time: float, mood: str) -> None:
+        self.name = str(name)
+        self.weight = float(weight)
+        # seconds
+        self.cooldown_time = float(cooldown_time)
+        self.mood = str(mood)
+
+
+class AnimationGroup:
+
+    __slots__ = [
+        "members"
+    ]
+
+    def __init__(self, members: Iterable[AnimationGroupMember]) -> None:
```

Comment:
Why `Iterable` and not directly a `Dict`? Would you like to support both lists and dictionaries for the animation group members?

Reply:
The members look like a list in animation group JSON files:
```
{
  "Animations": [
    {
      "Name": "anim_bored_01",
      "Weight": 1.0,
      "CooldownTime_Sec": 0.0,
      "Mood": "Default"
    },
    {
      "Name": "anim_bored_02",
      "Weight": 1.0,
      "CooldownTime_Sec": 0.0,
      "Mood": "Default"
    }
  ]
}
```

I don't mind changing to `Dict` if it is more convenient.

Reply:
Aaah, I get it now. I forgot that this is a list of items to choose randomly from. I'm good with Iterable, for sure, it was more a question to learn from :)

Reply:
Aaah, I get it now. I forgot that this is a list of items to choose randomly from. I'm good with Iterable, for sure, it was more a question to learn from :)